### ISSUE TITLE
Typescript 1.8.x

### ISSUE THREAD

**Issue by stefan-b314** [rollup/rollup-plugin-typescript#31]
Hi,

I would like to use your plugin with Typescript 1.8.x, so I tried

```
import typescript from 'rollup-plugin-typescript';
import * as ts from 'typescript';

export default {
    ...
    plugins: [
        typescript( {
            target: ts.ScriptTarget.ES6,
            typescript: ts
        } )
    ]
}
```

Unfortunately, there is a [change in the API](https://github.com/Microsoft/TypeScript/commit/1ee50223503fc646ab12bf191d0989760ec447c5) that is not compatible with your [current code](https://github.com/rollup/rollup-plugin-typescript/blob/0b14ac87faa7dbe6bd8541cad0f2866aa8f4d5a4/src/index.ts#L69), as the host is getting passed as the third parameter instead of the fourth.

What would be the best way to fix this?!
I can submit a PR, if you like... but wanted to discuss the best way before.

Kind regards,
Stefan

**Comment by Victorystick** [comment]
I'm not sure. I suppose we could check the TS version and act differently for each API. PRs are always welcome.

**Comment by RReverser** [comment]
Oh, so it's not my misconfiguration.

@stefanullinger care to send a PR?

I tried to patch code in node_modules locally to `var result = typescript.nodeModuleNameResolver(importee, importer, {}, resolveHost);` and that fixed the issue.

**Comment by stefan-b314** [comment]
Yes, I can create the PR in the next couple of hours.

**Comment by RReverser** [comment]
Another possibly related (1.8 changes) issue: `__awaiter` implementation differs in https://github.com/rollup/rollup-plugin-typescript/blob/master/src/typescript-helpers.ts#L22 and TS-generated code.

See https://github.com/tifosiblack/typescript-helpers/issues/2#issuecomment-192992043 for details.

**Comment by Victorystick** [comment]
Released as 0.6.0.
